Rename 'Open Distro' to follow open search naming convention

cliu123 opened the pull request:
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Maintenance
 
 
 2. __Github Issue # or road-map entry, if available:__



 3. __Description of changes:__

- Rename all formats of `Open Distro` to follow OpenSearch naming convention
- This is one of the changes required after OpenSearch beta1 release.


 4. __Why these changes are required?__ 
Migrating to OpenSearch


 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)
No changes in behavior


 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 UT
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)



 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)





By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

cliu123 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/WhitelistApiAction.java:
```diff
@@ -27,6 +27,7 @@
 import com.amazon.opendistroforelasticsearch.security.securityconf.impl.SecurityDynamicConfiguration;
 import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
 import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.amazon.opendistroforelasticsearch.security.tools.OpenSearchSecurityAdmin;
```
The `@link com.amazon.opendistroforelasticsearch.security.tools.OpenDistroSecurityAdmin` is not working without importing the class.

cliu123 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/NodesDnApiAction.java:
```diff
@@ -26,6 +26,7 @@
 import com.amazon.opendistroforelasticsearch.security.securityconf.impl.NodesDn;
 import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
 import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.amazon.opendistroforelasticsearch.security.tools.OpenSearchSecurityAdmin;
```
The @link com.amazon.opendistroforelasticsearch.security.tools.OpenDistroSecurityAdmin is currently not working without importing the class.

cliu123 commented on src/test/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/WhitelistApiTest.java:
```diff
@@ -21,6 +21,7 @@
 import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditMessage;
 import com.amazon.opendistroforelasticsearch.security.auditlog.integration.TestAuditlogImpl;
 import com.amazon.opendistroforelasticsearch.security.dlic.rest.validation.AbstractConfigurationValidator;
+import com.amazon.opendistroforelasticsearch.security.filter.OpenSearchSecurityRestFilter;
```
The `@link com.amazon.opendistroforelasticsearch.security.filter.OpenDistroSecurityRestFilter` is currently not working without importing the class.

cliu123 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/OpenSearchSecurityFlsDlsIndexSearcherWrapper.java:
```diff
@@ -33,14 +33,13 @@
 
 import com.amazon.opendistroforelasticsearch.security.auditlog.AuditLog;
 import com.amazon.opendistroforelasticsearch.security.compliance.ComplianceIndexingOperationListener;
-import com.amazon.opendistroforelasticsearch.security.privileges.PrivilegesEvaluator;
```
There're a few unused imports are deleted in this PR.

sujithvm commented:
Can we rename to more generic names and drop prefix `openSearch` unless it makes sense to have it. This should help if renaming conventions were to change again. For class the, package names should be able to make it distinguishable.  

For eg: 
openSearchSecurityIndex -> securityIndex
OpenSearchSecurityHttpServerTransport -> HttpServerTransport
OpenDistroSecurityUtils -> SecurityUtils  
OpenDistroSSLConfig ->SSLConfig

cliu123 commented:
> Can we rename to more generic names and drop prefix `openSearch` unless it makes sense to have it. This should help if renaming conventions were to change again. For classes the package names should be able to make it distinguishable.
> 
> For eg:
> openSearchSecurityIndex -> securityIndex
> OpenSearchSecurityHttpServerTransport -> HttpServerTransport
> OpenDistroSecurityUtils -> SecurityUtils
> OpenDistroSSLConfig ->SSLConfig

Distinguashing classes by package names could be confusing. The likelihood to rename these bunch of things again in the future is low enough not to worry about. I believe it will be fine to remain the same naming convention as before but renaming `opendistro` to `opensearch`. That has been working pretty well.

sujithvm commented:
@cliu123 Can you explain more why OpenSearch/openSearch prefix needs to be added/retained in class and variable names? 
Regarding the likelihood of naming conventions changing, we just had them changed 1.5 years back

cliu123 commented:
> @cliu123 Can you explain more why OpenSearch/openSearch prefix needs to be added/retained in class and variable names?
> Regarding the likelihood of naming conventions changing, we just had them changed 1.5 years back

@sujithvm The [requirement](https://quip-amazon.com/afeBAV0uMRlU/Plugin-renaming-check-list-in-OpenSearch-Draft#ENG9CA8iT8l) being shared cross all plugins of the migration is to replace `opendistro` with `opensearch` instead of removing the prefixes. Retaining the prefix will keep the consistency. Distinguishing classes with package names could be confusing. If retaining the `opensearch` prefix does not introduce significant risks/concerns, I would prefer to retain it. Otherwise, the entire PR would need to be redo. Find all&replace all don't work in this case because of the different format(e.g uppercase/lowercase) in the changes. It would required refactor each of the class, method and variables and all their usages to avoid compilation errors.

sujithvm commented:
> Retaining the prefix will keep the consistency.  

IMO they should not have been there in the first place and would prefer taking this opportunity to improve / remove whats unnecessary. 

> Distinguishing classes with package names could be confusing.  

Could you explain and give example of what is confusing ?

> Find all&replace all don't work in this case because of the different format 

Wont it be the same effort, find opendistro and replacing with empty string ? I agree the PR needs to be redone.

cliu123 commented:
> > Retaining the prefix will keep the consistency.
> 
> IMO they should not have been there in the first place and would prefer taking this opportunity to improve / remove whats unnecessary.
> 
> > Distinguishing classes with package names could be confusing.
> 
> Could you explain and give example of what is confusing ?
> 
> > Find all&replace all don't work in this case because of the different format
> 
> Wont it be the same effort, find opendistro and replacing with empty string ? I agree the PR needs to be redone.

cliu123 commented:
> > Can we rename to more generic names and drop prefix `openSearch` unless it makes sense to have it. This should help if renaming conventions were to change again. For classes the package names should be able to make it distinguishable.
> > For eg:
> > openSearchSecurityIndex -> securityIndex
> > OpenSearchSecurityHttpServerTransport -> HttpServerTransport
> > OpenDistroSecurityUtils -> SecurityUtils
> > OpenDistroSSLConfig ->SSLConfig
> 
> Distinguashing classes by package names could be confusing. The likelihood to rename these bunch of things again in the future is low enough not to worry about. I believe it will be fine to remain the same naming convention as before but renaming `opendistro` to `opensearch`. That has been working pretty well.

@saratvemulapalli Could you please help confirm if the [requirements](https://quip-amazon.com/afeBAV0uMRlU/Plugin-renaming-check-list-in-OpenSearch-Draft#ENG9CA8iT8l) requires all plugins to consistently follow this guidance? If not, I believe it should be fine to remove all the `OpenSearch`/`openSearch` prefix.

cliu123 commented:
> Can we rename to more generic names and drop prefix `openSearch` unless it makes sense to have it. This should help if renaming conventions were to change again. For classes the package names should be able to make it distinguishable.
> 
> For eg:
> openSearchSecurityIndex -> securityIndex
> OpenSearchSecurityHttpServerTransport -> HttpServerTransport
> OpenDistroSecurityUtils -> SecurityUtils
> OpenDistroSSLConfig ->SSLConfig

@saratvemulapalli Could you please help confirm if the requirements requires all plugins to consistently follow this guidance? If not, I believe it should be fine to remove all the OpenSearch/openSearch prefix.

saratvemulapalli commented:
> > Can we rename to more generic names and drop prefix `openSearch` unless it makes sense to have it. This should help if renaming conventions were to change again. For classes the package names should be able to make it distinguishable.
> > For eg:
> > openSearchSecurityIndex -> securityIndex
> > OpenSearchSecurityHttpServerTransport -> HttpServerTransport
> > OpenDistroSecurityUtils -> SecurityUtils
> > OpenDistroSSLConfig ->SSLConfig
> 
> @saratvemulapalli Could you please help confirm if the [requirements](https://quip-amazon.com/afeBAV0uMRlU/Plugin-renaming-check-list-in-OpenSearch-Draft#ENG9CA8iT8l) requires all plugins to consistently follow this naming convention? If not, we can think about the necessity to remove the prefixes.
> 
> > Distinguishing classes with package names could be confusing.
> 
> @sujithvm In a system with a lot of services integrated where security plugin is one of the services, classnames like `HttpServerTransport` and `SSLConfig` would not be that easy to locate with the classnames when debugging in logs.



> > Can we rename to more generic names and drop prefix `openSearch` unless it makes sense to have it. This should help if renaming conventions were to change again. For classes the package names should be able to make it distinguishable.
> > For eg:
> > openSearchSecurityIndex -> securityIndex
> > OpenSearchSecurityHttpServerTransport -> HttpServerTransport
> > OpenDistroSecurityUtils -> SecurityUtils
> > OpenDistroSSLConfig ->SSLConfig
> 
> @saratvemulapalli Could you please help confirm if the [requirements](https://quip-amazon.com/afeBAV0uMRlU/Plugin-renaming-check-list-in-OpenSearch-Draft#ENG9CA8iT8l) requires all plugins to consistently follow this naming convention? If not, we can think about the necessity to remove the prefixes.
> 
> > Distinguishing classes with package names could be confusing.
> 
> @sujithvm In a system with a lot of services integrated where security plugin is one of the services, classnames like `HttpServerTransport` and `SSLConfig` would not be that easy to locate with the classnames when debugging in logs.

Sure definitely. Lets discuss this in the issue: https://github.com/opensearch-project/opensearch-plugins/issues/13
My personal preference is to have OpenSearch in it. 

Let's not use internal links in the public, please use public issues to communicate. 
cc: @dblock

sujithvm commented:
IMO I don't find value in prefixing variable names (eg: securityIndex -> opensearchSecurityIndex) and class names (eg: SecurityUtils -> OpenSearchSecurityUtils). If it was already done that way, wanted to see if we could use this opportunity to clean them up. A lot of rebranding effort happened just 1.5years ago from SearchGuard. 

If this is a blocker going forward for any release, please go ahead with this PR.

vrozov reviewed (changes requested):
It is not OK to move forward with less readable code even if it blocks a release. The code needs to be readable and maintainable.

andy840314 commented on securityconfig/opensearch.yml.example:
```diff
@@ -37,10 +37,10 @@ opendistro_security.nodes_dn_dynamic_config_enabled: false
 # MAPPING_ONLY - mappings must be configured explicitely in roles_mapping.yml (default)
-# BACKENDROLES_ONLY - backend roles are mapped to Open Distro Security rules directly. Settings in roles_mapping.yml have no effect.
-# BOTH - backend roles are mapped to Open Distro Security roles mapped directly and via roles_mapping.yml in addition
+# BACKENDROLES_ONLY - backend roles are mapped to OpenSearch Security rules directly. Settings in roles_mapping.yml have no effect.
+# BOTH - backend roles are mapped to Security roles mapped directly and via roles_mapping.yml in addition
```
Why some of the `Open Distro` are deleted, some are renamed to `OpenSearch`? Also `Security rules` seems to be a typo for `Security roles`.

cliu123 replied:
Updated

andy840314 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java:
```diff
@@ -87,7 +87,7 @@
     private final Settings settings;
     private volatile AuditConfig.Filter auditConfigFilter;
-    private final String opendistrosecurityIndex;
+    private final String openSearchSecurityIndex;
```
Should we use `securityIndex` here?

andy840314 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/WhitelistApiAction.java:
```diff
@@ -27,6 +27,7 @@
 import com.amazon.opendistroforelasticsearch.security.securityconf.impl.SecurityDynamicConfiguration;
 import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
 import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.amazon.opendistroforelasticsearch.security.tools.SecurityAdmin;
```
Import unused.

cliu123 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/RequestResolver.java:
```diff
@@ -85,7 +85,7 @@
             final boolean resolveIndices,
             final boolean resolveBulk,
-            final String opendistrosecurityIndex,
+            final String openSearchsecurityIndex,
```
Done

cliu123 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/OpenSearchSecurityPlugin.java:
```diff
@@ -50,18 +50,21 @@
 import com.amazon.opendistroforelasticsearch.security.compliance.ComplianceIndexingOperationListenerImpl;
-import com.amazon.opendistroforelasticsearch.security.configuration.DlsFlsValveImpl;
-import com.amazon.opendistroforelasticsearch.security.configuration.OpenDistroSecurityFlsDlsIndexSearcherWrapper;
+import com.amazon.opendistroforelasticsearch.security.configuration.*;
```
> I don't think we should use *. Seems that IntelliJ do this by default.
Done.

cliu123 replied:
> May be can we add a linter to take care of this?
> And run the linter check in the CI's.
> 
> Other plugins use spotless you can explore different solutions if you are not on gradle yet.

Sure. We can try to find a similar solution and add it with separate PRs.

cliu123 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/RequestResolver.java:
```diff
@@ -172,7 +172,7 @@ private static AuditMessage resolveInner(final AuditCategory category,
             final boolean logRequestBody,
             final boolean resolveIndices,
-            final String opendistrosecurityIndex,
+            final String openSearchsecurityIndex,
```
Done

andy840314 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/RequestResolver.java:
```diff
@@ -349,7 +349,7 @@ private static void addIndicesSourceSafe(final AuditMessage msg,
             final boolean addSource,
             final boolean sourceIsSensitive,
-            final String opendistrosecurityIndex) {
+            final String openSearchSecurityIndex) {
```
Sorry, missed this variable. Need to rename this too.

cliu123 replied:
Done

andy840314 commented on src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/OpenSearchSecuritySSLPlugin.java:
```diff
@@ -99,17 +99,17 @@
     protected final Settings settings;
     protected final SharedGroupFactory sharedGroupFactory;
-    protected final OpenDistroSecurityKeyStore odsks;
+    protected final SecurityKeyStore odsks;
```
Should this be renamed to `sks`?

cliu123 replied:
Done

vrozov reviewed (dismissed):
- class renames needs to be combined with package renames to avoid several PRs with large number of files modified and that invalidate all other PRs.
- there are renames that do not apply, for example there is no OpenSearch 6.x.

cliu123 commented:
> * class renames needs to be combined with package renames to avoid several PRs with large number of files modified and that invalidate all other PRs.

It is easier to review the PRs with fewer number of files separately, but they can be merged together.

> * there are renames that do not apply, for example there is no OpenSearch 6.x.

Updated

vrozov commented:
> > * class renames needs to be combined with package renames to avoid several PRs with large number of files modified and that invalidate all other PRs.
> 
> It is easier to review the PRs with fewer number of files separately, but they can be merged together.
> 

I don't see why it will be easier to review 2 PRs with 400 files and 100 files changed, instead of 1 PR with 400 files changed. Package change is a rename, so it can be combined with class rename and other logical changes can be done in a separate PR.

saratvemulapalli commented on src/main/java/org/opensearch/security/auditlog/impl/AbstractAuditLog.java:
```diff
@@ -426,7 +426,7 @@ public void logDocumentWritten(ShardId shardId, GetResult originalResult, Index
         }
 
-        AuditCategory category = opendistrosecurityIndex.equals(shardId.getIndexName())? AuditCategory.COMPLIANCE_INTERNAL_CONFIG_WRITE: AuditCategory.COMPLIANCE_DOC_WRITE;
+        AuditCategory category = securityIndex.equals(shardId.getIndexName())? AuditCategory.COMPLIANCE_INTERNAL_CONFIG_WRITE: AuditCategory.COMPLIANCE_DOC_WRITE;
```
I like this!

saratvemulapalli reviewed (approved):
The changes look good to me!

hsiang9431-amzn commented on src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java:
```diff
@@ -99,17 +100,17 @@
     protected final Path configPath;
     private final static SslExceptionHandler NOOP_SSL_EXCEPTION_HANDLER = new SslExceptionHandler() {};
-    protected final OpenDistroSSLConfig openDistroSSLConfig;
+    protected final org.opensearch.security.ssl.transport.SSLConfig SSLConfig;
```
should be better:
```java
import org.opensearch.security.ssl.transport.SSLConfig;
```

cliu123 replied:
It's imported on Line#23

hsiang9431-amzn reviewed (changes requested):
prevent using full class path in code

andy840314 commented on src/main/java/org/opensearch/security/dlic/rest/api/NodesDnApiAction.java:
```diff
@@ -52,10 +53,10 @@
  *
  * The backing data is stored in {@link ConfigConstants#OPENDISTRO_SECURITY_CONFIG_INDEX_NAME} which is populated during bootstrap.
- * For existing clusters, {@link com.amazon.opendistroforelasticsearch.security.tools.OpenDistroSecurityAdmin} tool can
+ * For existing clusters, {@link SecurityAdmin} tool can
  * be used to populate the index.
  *
- * See {@link com.amazon.opendistroforelasticsearch.security.dlic.rest.api.NodesDnApiTest} for usage examples.
+ * See {@link org.opensearch.security.dlic.rest.api.NodesDnApiTest} for usage examples.
```
Should we give the full class name or not? line 56, 59 is different.

cliu123 replied:
We actually don't need the full path. I've changed the line 56.

cliu123 commented:
> It is not OK to move forward with less readable code even if it blocks a release. The code needs to be readable and maintainable.

Removed the prefixes.

cliu123 commented:
> * class renames needs to be combined with package renames to avoid several PRs with large number of files modified and that invalidate all other PRs.
> * there are renames that do not apply, for example there is no OpenSearch 6.x.

- Moved the namespace changes to this PR.
- Reverted the change.

andy840314 commented on src/test/java/org/opensearch/security/AggregationTests.java:
```diff
@@ -42,9 +42,8 @@
 
-import com.amazon.opendistroforelasticsearch.security.test.SingleClusterTest;
-import com.amazon.opendistroforelasticsearch.security.test.helper.rest.RestHelper;
-import com.amazon.opendistroforelasticsearch.security.test.helper.rest.RestHelper.HttpResponse;
```
This should be reverted. There are many other files that have that removes
`import com.amazon.opendistroforelasticsearch.security.test.helper.rest.RestHelper.HttpResponse;`
Should add this back.

cliu123 replied:
Done

cliu123 replied:
Done.

andy840314 commented on src/test/java/org/opensearch/security/HttpIntegrationTests.java:
```diff
@@ -564,9 +563,9 @@ public void testBulkWithOneIndexFailure() throws Exception {
+        RestHelper.HttpResponse res = rh.executePostRequest("_bulk?refresh=true", bulkBody, encodeBasicHeader("bulk_test_user", "nagilum"));
         System.out.println(res.getBody());
-        JsonNode jsonNode = readTree(res.getBody());
+        JsonNode jsonNode = DefaultObjectMapper.readTree(res.getBody());
```
Should revert this.

cliu123 replied:
Done.

andy840314 commented on src/test/java/org/opensearch/security/auditlog/compliance/ComplianceConfigTest.java:
```diff
@@ -45,8 +46,8 @@ public void testDefault() {
         assertFalse(complianceConfig.shouldLogWriteMetadataOnly());
         assertFalse(complianceConfig.shouldLogDiffsForWrite());
-        assertEquals(defaultIgnoredUserMatcher, complianceConfig.getIgnoredComplianceUsersForReadMatcher());
-        assertEquals(defaultIgnoredUserMatcher, complianceConfig.getIgnoredComplianceUsersForWriteMatcher());
+        Assert.assertEquals(defaultIgnoredUserMatcher, complianceConfig.getIgnoredComplianceUsersForReadMatcher());
+        Assert.assertEquals(defaultIgnoredUserMatcher, complianceConfig.getIgnoredComplianceUsersForWriteMatcher());
```
Revert this.

cliu123 replied:
Done.

andy840314 commented on src/test/java/org/opensearch/security/auditlog/compliance/ComplianceConfigTest.java:
```diff
@@ -78,8 +79,8 @@ public void testConfig() {
         assertTrue(complianceConfig.shouldLogWriteMetadataOnly());
         assertFalse(complianceConfig.shouldLogDiffsForWrite());
-        assertEquals(WildcardMatcher.from(ImmutableSet.of("test-user-1", "test-user-2")), complianceConfig.getIgnoredComplianceUsersForReadMatcher());
-        assertEquals(WildcardMatcher.from(ImmutableSet.of("test-user-3", "test-user-4")), complianceConfig.getIgnoredComplianceUsersForWriteMatcher());
+        Assert.assertEquals(WildcardMatcher.from(ImmutableSet.of("test-user-1", "test-user-2")), complianceConfig.getIgnoredComplianceUsersForReadMatcher());
+        Assert.assertEquals(WildcardMatcher.from(ImmutableSet.of("test-user-3", "test-user-4")), complianceConfig.getIgnoredComplianceUsersForWriteMatcher());
```
Revert this.

cliu123 replied:
Done.

andy840314 commented on src/test/java/org/opensearch/security/auditlog/compliance/ComplianceConfigTest.java:
```diff
@@ -13,13 +13,14 @@
+import org.opensearch.security.support.WildcardMatcher;
 
 import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
```
Revert this.

cliu123 replied:
Done.

andy840314 commented on src/test/java/org/opensearch/security/auditlog/config/AuditConfigFilterTest.java:
```diff
@@ -13,24 +13,18 @@
+import org.opensearch.security.auditlog.impl.AuditCategory;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.support.WildcardMatcher;
+import org.junit.Assert;
```
There are lots of test class that add `import org.junit.Assert;` which is unnecessary.

andy840314 commented on src/test/java/org/opensearch/security/auditlog/impl/AuditCategoryTest.java:
```diff
@@ -48,11 +45,11 @@ public AuditCategoryParseTest(List<String> input, EnumSet<AuditCategory> expecte
+                    {Arrays.asList("BAD_HEADERS"), EnumSet.of(AuditCategory.BAD_HEADERS)},
+                    {Arrays.asList("bad_headers"), EnumSet.of(AuditCategory.BAD_HEADERS)},
+                    {Arrays.asList("bAd_HeAdErS"), EnumSet.of(AuditCategory.BAD_HEADERS)},
+                    {Arrays.asList("bAd_HeAdErS"), EnumSet.of(AuditCategory.BAD_HEADERS)},
+                    {Arrays.asList("BAD_HEADERS", "AUTHENTICATED"), EnumSet.of(AuditCategory.BAD_HEADERS, AuditCategory.AUTHENTICATED)},
```
Not sure if we have to do this.

cliu123 replied:
Reverted

andy840314 commented on src/test/java/org/opensearch/security/auth/RolesInjectorTest.java:
```diff
@@ -44,7 +43,7 @@ public void testNotInjected() {
     public void testInjected() {
         ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
-        threadContext.putTransient(OPENDISTRO_SECURITY_INJECTED_ROLES, "user1|role_1,role_2");
+        threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROLES, "user1|role_1,role_2");
```
Not sure if the change is necessary.

cliu123 replied:
Reverted

andy840314 commented on src/test/java/org/opensearch/security/auth/RolesInjectorTest.java:
```diff
@@ -69,7 +68,7 @@ public void testCorruptedInjection() {
         corruptedStrs.forEach(name -> {
             ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
-            threadContext.putTransient(OPENDISTRO_SECURITY_INJECTED_ROLES, name);
+            threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROLES, name);
```
Not sure if the change is necessary.

cliu123 replied:
Reverted

andy840314 commented on src/test/java/org/opensearch/security/configuration/SaltTest.java:
```diff
@@ -39,8 +39,8 @@ public void testDefault() {
         // assert
-        assertEquals(SALT_SIZE, salt.getSalt16().length);
-        assertArrayEquals(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_SALT_DEFAULT.getBytes(StandardCharsets.UTF_8), salt.getSalt16());
+        Assert.assertEquals(Salt.SALT_SIZE, salt.getSalt16().length);
```
Not sure if the change is necessary.

cliu123 replied:
Reverted

andy840314 commented on src/test/java/org/opensearch/security/dlic/rest/api/AuditApiActionTest.java:
```diff
@@ -109,7 +108,7 @@ public void testDisabledCategoryOrder() throws Exception {
         testPutRequest(json, HttpStatus.SC_OK, true);
         RestHelper.HttpResponse response = rh.executeGetRequest(ENDPOINT, adminCredsHeader);
-        List<String> actual = Streams.stream(readTree(response.getBody()).at("/config/audit/disabled_rest_categories").iterator())
+        List<String> actual = Streams.stream(DefaultObjectMapper.readTree(response.getBody()).at("/config/audit/disabled_rest_categories").iterator())
```
Not sure if the change is necessary.

andy840314 replied:
Same issue at other lines in this file.

cliu123 replied:
Reverted

andy840314 commented on src/test/java/org/opensearch/security/dlic/rest/api/AuditApiActionTest.java:
```diff
@@ -125,7 +124,7 @@ public void testInvalidDisabledCategories() throws Exception {
         ), ComplianceConfig.DEFAULT);
         ObjectNode json = DefaultObjectMapper.objectMapper.valueToTree(auditConfig);
-        RestHelper.HttpResponse response = rh.executePutRequest(CONFIG_ENDPOINT, writeValueAsString(json, false));
+        RestHelper.HttpResponse response = rh.executePutRequest(CONFIG_ENDPOINT, DefaultObjectMapper.writeValueAsString(json, false));
```
Not sure if the change is necessary.

andy840314 commented on src/test/java/org/opensearch/security/dlic/rest/api/AuditApiActionTest.java:
```diff
@@ -134,7 +133,7 @@ public void testInvalidDisabledCategories() throws Exception {
         ), ComplianceConfig.DEFAULT);
         json = DefaultObjectMapper.objectMapper.valueToTree(auditConfig);
-        response = rh.executePutRequest(CONFIG_ENDPOINT, writeValueAsString(json, false));
+        response = rh.executePutRequest(CONFIG_ENDPOINT, DefaultObjectMapper.writeValueAsString(json, false));
```
Not sure if the change is necessary.

andy840314 commented on src/test/java/org/opensearch/security/dlic/rest/api/AuditApiActionTest.java:
```diff
@@ -371,15 +370,15 @@ private void testPutAction(final String payload, final int expectedStatus, final
         if (expectedStatus == HttpStatus.SC_OK) {
             response = rh.executeGetRequest(ENDPOINT, headers);
-            assertEquals(readTree(payload), readTree(response.getBody()).get("config"));
+            Assert.assertEquals(DefaultObjectMapper.readTree(payload), DefaultObjectMapper.readTree(response.getBody()).get("config"));
```
Same issue mention before. Regarding `Assert`

cliu123 replied:
Reverted

andy840314 commented on src/test/java/org/opensearch/security/support/Base64HelperTest.java:
```diff
@@ -32,17 +32,14 @@
 
     private static Serializable ds(Serializable s) {
-        return deserializeObject(serializeObject(s));
+        return Base64Helper.deserializeObject(Base64Helper.serializeObject(s));
```
Not sure if the change is necessary.

andy840314 replied:
Same issue with other lines in this file

cliu123 replied:
Reverted

andy840314 commented on src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java:
```diff
@@ -198,7 +199,7 @@ protected AbstractConfigurationValidator postProcessApplyPatchResult(RestChannel
             }
 
             ((ObjectNode) updatedResourceAsJsonNode).remove("password");
-            ((ObjectNode) updatedResourceAsJsonNode).set("hash", new TextNode(hash(plainTextPassword.toCharArray())));
+            ((ObjectNode) updatedResourceAsJsonNode).set("hash", new TextNode(Utils.hash(plainTextPassword.toCharArray())));
```
Revert this.

andy840314 commented on src/main/java/org/opensearch/security/filter/SecurityFilter.java:
```diff
@@ -257,7 +257,7 @@ public int order() {
             }
 
-            if(Origin.LOCAL.toString().equals(threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN))
+            if(AuditLog.Origin.LOCAL.toString().equals(threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN))
```
Revert this.

andy840314 commented on src/test/java/org/opensearch/security/EncryptionInTransitMigrationTests.java:
```diff
@@ -43,7 +42,7 @@ private void testSslOnlyMode(boolean dualModeEnabled) throws Exception {
         final RestHelper rh = nonSslRestHelper();
 
-        HttpResponse res = rh.executeGetRequest("_opendistro/_security/sslinfo");
+        RestHelper.HttpResponse res = rh.executeGetRequest("_opendistro/_security/sslinfo");
```
Revert this and line 102, 115

andy840314 commented on src/test/java/org/opensearch/security/IntegrationTests.java:
```diff
@@ -879,7 +879,7 @@ public void testSecurityIndexSecurity() throws Exception {
                 + "{ \"delete\" : { \"_index\" : \".opendistro_security\", \"_id\" : \"config\" } }\n";
         res = rh.executePostRequest("_bulk?refresh=true&pretty", bulkBody, encodeBasicHeader("nagilum", "nagilum"));
-        JsonNode jsonNode = readTree(res.getBody());
+        JsonNode jsonNode = DefaultObjectMapper.readTree(res.getBody());
```
Revert this.

cliu123 replied:
Done

andy840314 commented on src/test/java/org/opensearch/security/auditlog/config/AuditConfigFilterTest.java:
```diff
@@ -45,7 +45,7 @@ public void testDefault() {
         // act
         final AuditConfig.Filter auditConfigFilter = AuditConfig.Filter.from(Settings.EMPTY);
-        // assert
+        // asser
```
Revert this.

cliu123 replied:
Done

andy840314 commented on src/test/java/org/opensearch/security/auditlog/config/AuditConfigSerializeTest.java:
```diff
@@ -13,18 +13,19 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
```
Revert this and line 39, 40, 184, 277, 278, 330, 331

cliu123 replied:
Done

andy840314 commented on src/test/java/org/opensearch/security/configuration/SaltTest.java:
```diff
@@ -13,9 +13,10 @@
 
-import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.opensearch.security.support.ConfigConstants;
+import org.junit.Assert;
```
Revert this.

cliu123 replied:
Done

andy840314 commented on src/test/java/org/opensearch/security/dlic/rest/api/AuditApiActionTest.java:
```diff
@@ -31,6 +30,7 @@
 import com.google.common.collect.Streams;
 import org.apache.http.Header;
 import org.apache.http.HttpStatus;
+import org.junit.Assert;
```
Revert this and line 508, 558

cliu123 replied:
Done
